internal/controller/utils.go

```diff
@@ -149,6 +149,17 @@ func createOrPatchStatefulSet(ctx context.Context, logger logr.Logger, ec *ecv1a
 		},
 	}
 
+	if len(ec.Spec.EtcdOptions) > 0 {
+		for i := range podSpec.Containers {
+			if podSpec.Containers[i].Name == "etcd" {
+				for _, v := range ec.Spec.EtcdOptions {
+					podSpec.Containers[i].Env = append(podSpec.Containers[i].Env, corev1.EnvVar{Name: v.Name, Value: v.Value})
```

Comment:
I think we have two options:
- append more args directly (see line 103)
- use env variable as this PR does

Either way, it'd be better to let users to configure the item as the flag (i.e. "`data-dir`"),  and then,
- either append it to the container's Args,
- or convert it into environment (adding prefix "ETCD_"). refer to https://etcd.io/docs/v3.6/op-guide/configuration/

Which one do you prefer? Personally, the first option seems more direct and simpler, so better?

Reply:
If you go for option 2, then you need to convert the flag to env variable, 

refer to https://github.com/etcd-io/etcd/blob/ffb6ff9acf049b6b864955bea46ac6e33c54d9c9/pkg/flags/flag.go#L69-L71

The prefix is `ETCD`.

Reply:
If users provide them as flags then I would just go with appending the args array, don’t see the point of converting to env vars.

I’ll make the update when able.

Reply:
@ahrtr I've made changes to use args instead of env vars and added some unit tests for the additional functions.